tasks/Functions-admin/deployFullOracle.js

```diff
@@ -22,33 +22,40 @@ task(
-  const oracleFactory = overrides ? await oracleFactoryFactory.deploy(overrides) : await oracleFactoryFactory.deploy()
-  console.log(`Waiting for transaction ${oracleFactory.deployTransaction.hash} to be confirmed...`)
-  await oracleFactory.deployTransaction.wait(2)
-  console.log(`FunctionsOracleFactory deployed to ${oracleFactory.address} on ${network.name}`)
-
-  console.log("Deploying Functions oracle")
-  const accounts = await ethers.getSigners()
-  const deployer = accounts[0]
-  const OracleDeploymentTransaction = overrides
-    ? await oracleFactory.deployNewOracle(overrides)
-    : await oracleFactory.deployNewOracle()
-  console.log(`Waiting for transaction ${OracleDeploymentTransaction.hash} to be confirmed...`)
-  const OracleDeploymentReceipt = await OracleDeploymentTransaction.wait(1)
-  const FunctionsOracleAddress = OracleDeploymentReceipt.events[1].args.don
-  const oracle = await ethers.getContractAt("FunctionsOracle", FunctionsOracleAddress, deployer)
-  console.log(`FunctionsOracle deployed to ${oracle.address} on ${network.name}`)
+  const proxyAdminAddress = await upgrades.deployProxyAdmin()
+  console.log(`Proxy Admin contract is deployed at ${proxyAdminAddress} on ${network.name}`)
+  const proxyAdmin = await ethers.getContractAt("ProxyAdmin", proxyAdminAddress, await ethers.getSigner())
+
+  let oracleProxy
+  const FunctionsOracle = await ethers.getContractFactory("FunctionsOracle")
+  console.log("Deploying Functions Oracle Proxy & implementation contract")
+  oracleProxy = overrides
+    ? await upgrades.deployProxy(FunctionsOracle, [], {
+        kind: "transparent",
+        constructorArgs: overrides,
+      })
+    : await upgrades.deployProxy(FunctionsOracle)
+  console.log(`Waiting for the transaction ${oracleProxy.deployTransaction.hash} to be confirmed`)
+  await oracleProxy.deployTransaction.wait(1)
+  const oracleImplementation = await proxyAdmin.getProxyImplementation(oracleProxy.address)
+  console.log(
+    `FunctionsOracle proxy deployed to ${oracleProxy.address} using the implementation at ${oracleImplementation} on ${network.name}`
+  )
 
   console.log("Deploying Functions registry", linkEthFeedAddress, linkTokenAddress)
   const registryFactory = await ethers.getContractFactory("FunctionsBillingRegistry")
-  const registry = overrides
-    ? await registryFactory.deploy(linkTokenAddress, linkEthFeedAddress, oracle.address, overrides)
-    : await registryFactory.deploy(linkTokenAddress, linkEthFeedAddress, oracle.address)
-  console.log(`Waiting for transaction ${registry.deployTransaction.hash} to be confirmed...`)
-  await registry.deployTransaction.wait(1)
-  console.log(`FunctionsBillingRegistry deployed to ${registry.address} on ${network.name}`)
+  const registryProxy = overrides
+    ? await upgrades.deployProxy(registryFactory, [linkTokenAddress, linkEthFeedAddress, oracleProxy.address], {
+        kind: "transparent",
+        constructorArgs: overrides,
```

Comment:
`constructorArgs` seems like a weird name, but I assume this is just forwarding the overrides to the underlying `ethers` call.
Just double-checking this is the correct way to specify overrides.

Reply:
I know what you mean, It'll get tested it when I do the Goerli deployment. If it isn't `constructorArgs` then I'm not sure that Hardhat Upgrades exposes a way to override deployment settings.

Reply:
Thanks for double-checking!

Reply:
Removed these, they indeed do not work.